### PR TITLE
Clarify which node / resource types support `node_color`

### LINKED ISSUE #4492: Update docs for `node_color`
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/resource-configs/docs

### What part(s) of the page would you like to see updated?

Two things:
1. Expand code examples across all applicable node types
2. (Option) Consider more explicit language of which node types support `node_color` (or not)

### More explanation

Only the model node type includes `node_color` within its general code [example](https://docs.getdbt.com/reference/resource-configs/docs):

image

And [elsewhere](https://docs.getdbt.com/reference/resource-configs/docs#definition), it only says "some node types" rather than being explicit about which support "node_color" and which don't:

image

### Additional information

`node_color` was added within https://github.com/dbt-labs/dbt-core/pull/5397

See https://github.com/dbt-labs/dbt-core/issues/8061#issuecomment-1863292897 for an adjacent issue.

### PR DESCRIPTION
**Pull request by mirnawong1**
clarifies which node type supports `node_color`. Resolves #4492

### CODE REVIEW

**Comment by mirnawong1** [comment]
hey @dbeatty10 ! could i ask you to review this pr to make sure it addresses your issues?

**Comment by mirnawong1** [comment]
hey @dbeatty10 wanted to check in on what your thoughts were on this pr?

File: website/docs/reference/resource-configs/docs.md [hunk 1]
```diff
@@ -156,7 +156,7 @@ macros:
 </Tabs>
 
 ## Definition
-The docs field can be used to provide documentation-specific configuration to models. It supports the doc attribute `show`, which controls whether or not models are shown in the auto-generated documentation website. It also supports `node_color` for some node types.
+The docs field can be used to provide documentation-specific configuration to models. It supports the doc attribute `show`, which controls whether or not models are shown in the auto-generated documentation website. It also supports `node_color` for the node type "Model".
```
Comment (dbeatty10) [inline website/docs/reference/resource-configs/docs.md]: Here's the ones that work (or don't): https://github.com/dbt-labs/docs.getdbt.com/issues/4492#issuecomment-1881658603

```suggestion
The docs field can be used to provide documentation-specific configuration to models. It supports the doc attribute `show`, which controls whether or not models are shown in the auto-generated documentation website. It also supports `node_color` for models, seeds, snapshots, and analyses. Other node types are not supported.
```

**Review by dbeatty10 — COMMENTED** [review]
Could we add example code for `node_color: color_id` for each of these node types?
- models
- seeds
- snapshots
- analyses

Per https://github.com/dbt-labs/docs.getdbt.com/issues/4492#issuecomment-1881658603, these are the (4) node / resource types that support using `node_color`.

And also make sure the example configuration appears in _both_ of these file examples for each node type?
- `dbt_project.yml`
- `schema.yml`

So (8) places total 😅

Here's the only one out of the eight that exists currently:

`models/schema.yml`

```
version: 2

models:
- name: model_name
  docs:
    show: true | false
    node_color: "black"
```

To be consistent with example code for other configs/properties, we'd probably want to change `"black"` to a placeholder like `color_id`.

**Comment by mirnawong1** [comment]
hey @dbeatty10 amazing feedback! i've added it here and folded it all in. one thing i think we should add:

- it looks like the ide dag doesn't reflect the node color changes and you need to run dbt docs generate to see them. is that right? if so, it's pretty manual and i wonder if i should let product know and also flag this in the docs that they should run the command to see it.

File: website/docs/reference/resource-configs/docs.md [hunk 2]
```diff
@@ -206,7 +211,7 @@ models:
 
 The `docs` attribute now supports `node_color` to customize the display color of some node types in the DAG within dbt docs. You can define node colors in the files below and apply overrides where needed.
 
-`node_color` hiearchy:
+`node_color` hierarchy:
```
Comment (dbeatty10) [inline website/docs/reference/resource-configs/docs.md:214]: Eagle eyes! 🦅

File: website/docs/reference/analysis-properties.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ analyses:
     [description](/reference/resource-properties/description): <markdown_string>
     [docs](/reference/resource-configs/docs):
       show: true | false
+      node_color: color_id # Use name (such as node_color: purple) or hex code with quotes (such as node_color: "#cd7f32")
```
Comment (dbeatty10) [inline website/docs/reference/analysis-properties.md]: Suggesting using `<>` for consistency:

```suggestion
      node_color: <color_id> # Use name (such as node_color: purple) or hex code with quotes (such as node_color: "#cd7f32")
```

File: website/docs/reference/model-properties.md [hunk 4]
```diff
@@ -16,6 +16,7 @@ models:
     [description](/reference/resource-properties/description): <markdown_string>
     [docs](/reference/resource-configs/docs):
       show: true | false
+      node_color: color_id # Use name (such as node_color: purple) or hex code with quotes (such as node_color: "#cd7f32")
```
Comment (dbeatty10) [inline website/docs/reference/model-properties.md]: Suggesting using `<>` for consistency:

```suggestion
      node_color: <color_id> # Use name (such as node_color: purple) or hex code with quotes (such as node_color: "#cd7f32")
```

File: website/docs/reference/seed-properties.md [hunk 5]
```diff
@@ -16,6 +16,7 @@ seeds:
     [description](/reference/resource-properties/description): <markdown_string>
     [docs](/reference/resource-configs/docs):
       show: true | false
+      node_color: color_id # Use name (such as node_color: purple) or hex code with quotes (such as node_color: "#cd7f32")
```
Comment (dbeatty10) [inline website/docs/reference/seed-properties.md]: Suggesting using `<>` for consistency:

```suggestion
      node_color: <color_id> # Use name (such as node_color: purple) or hex code with quotes (such as node_color: "#cd7f32")
```

File: website/docs/reference/snapshot-properties.md [hunk 6]
```diff
@@ -20,6 +20,7 @@ snapshots:
     [meta](/reference/resource-configs/meta): {<dictionary>}
     [docs](/reference/resource-configs/docs):
       show: true | false
+      node_color: color_id # Use name (such as node_color: purple) or hex code with quotes (such as node_color: "#cd7f32")
```
Comment (dbeatty10) [inline website/docs/reference/snapshot-properties.md]: Suggesting using `<>` for consistency:

```suggestion
      node_color: <color_id> # Use name (such as node_color: purple) or hex code with quotes (such as node_color: "#cd7f32")
```

**Review by dbeatty10 — COMMENTED** [review]
Made the same suggestion (4) times -- once for each of the (4) `*-properties.md` files.

Didn't make the same suggestion for `resource-configs/docs.md` because it isn't currently using the `<...>` convention.

**Comment by dbeatty10** [comment]
> one thing i think we should add:
> 
> * it looks like the ide dag doesn't reflect the node color changes and you need to run dbt docs generate to see them. is that right? if so, it's pretty manual and i wonder if i should let product know and also flag this in the docs that they should run the command to see it.

Good catch! Makes sense to do both of these:
1. Add any relevant instructions how to make any changes display (i.e., re-run `dbt docs generate`)
1. Let product know

**Review by dbeatty10 — APPROVED** [review]

**Pull request merged**